src/components/navigation/navigation.component.jsx

```diff
@@ -0,0 +1,34 @@
+import React from "react";
+//Styles
+import "./navigation.styles.scss";
+// Components
+import NavigationDesktop from "../navigation-desktop/navigation-desktop.component.jsx";
+import NavigationMobile from "../navigation-mobile/navigation-mobile.component.jsx";
+
+class Navigation extends React.Component {
+    constructor() {
+        super();
+        this.state = {
+            width: window.innerWidth,
+        };
+        this.handleWindowSizeChange = this.handleWindowSizeChange.bind(this);
+    }
+    UNSAFE_componentWillMount() {
+        window.addEventListener("resize", this.handleWindowSizeChange);
```

Comment:
This method was deprecated for various reasons and as the first chunk of the name suggest, it is unsafe to use. Is it necessary to call adding new event listener in this LC method, won't componentDidMount work or If you really have to call it before mount maybe wrap it inside constructor?